hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/CockroachLegacyDialect.java

```diff
@@ -0,0 +1,805 @@
+		// Support was added in 20.1: https://www.cockroachlabs.com/docs/v20.1/select-for-update.html
+		if ( getVersion().isBefore( 20, 1 ) ) {
+			return "";
+		}
+		/*
+		 * Parent's implementation for (aliases, lockOptions) ignores aliases.
+		 */
+		if ( aliases.isEmpty() ) {
+			LockMode lockMode = lockOptions.getLockMode();
+			final Iterator<Map.Entry<String, LockMode>> itr = lockOptions.getAliasLockIterator();
+			while ( itr.hasNext() ) {
+				// seek the highest lock mode
+				final Map.Entry<String, LockMode> entry = itr.next();
+				final LockMode lm = entry.getValue();
+				if ( lm.greaterThan( lockMode ) ) {
+					aliases = entry.getKey();
+				}
+			}
+		}
+		LockMode lockMode = lockOptions.getAliasSpecificLockMode( aliases );
+		if (lockMode == null ) {
+			lockMode = lockOptions.getLockMode();
+		}
+		switch ( lockMode ) {
+			case PESSIMISTIC_READ: {
+				return getReadLockString( aliases, lockOptions.getTimeOut() );
+			}
+			case PESSIMISTIC_WRITE: {
+				return getWriteLockString( aliases, lockOptions.getTimeOut() );
+			}
+			case UPGRADE_NOWAIT:
+			case PESSIMISTIC_FORCE_INCREMENT: {
+				return getForUpdateNowaitString(aliases);
+			}
+			case UPGRADE_SKIPLOCKED: {
+				return getForUpdateSkipLockedString(aliases);
+			}
+			default: {
+				return "";
+			}
+		}
+	}
+
+	private String withTimeout(String lockString, int timeout) {
+		switch (timeout) {
+			case LockOptions.NO_WAIT: {
+				return supportsNoWait() ? lockString + " nowait" : lockString;
+			}
+			case LockOptions.SKIP_LOCKED: {
+				return supportsSkipLocked() ? lockString + " skip locked" : lockString;
+			}
+			default: {
+				return lockString;
+			}
+		}
+	}
+
+	@Override
+	public String getWriteLockString(int timeout) {
+		return withTimeout( getForUpdateString(), timeout );
+	}
+
+	@Override
+	public String getWriteLockString(String aliases, int timeout) {
+		return withTimeout( getForUpdateString( aliases ), timeout );
+	}
+
+	@Override
+	public String getReadLockString(int timeout) {
+		return withTimeout(" for share", timeout );
+	}
+
+	@Override
+	public String getReadLockString(String aliases, int timeout) {
+		return withTimeout(" for share of " + aliases, timeout );
+	}
+
+	@Override
+	public String getForUpdateNowaitString() {
+		return supportsNoWait()
+				? " for update nowait"
+				: getForUpdateString();
+	}
+
+	@Override
+	public String getForUpdateNowaitString(String aliases) {
+		return supportsNoWait()
+				? " for update of " + aliases + " nowait"
+				: getForUpdateString(aliases);
+	}
+
+	@Override
+	public String getForUpdateSkipLockedString() {
+		return supportsSkipLocked()
+				? " for update skip locked"
+				: getForUpdateString();
+	}
+
+	@Override
+	public String getForUpdateSkipLockedString(String aliases) {
+		return supportsSkipLocked()
+				? " for update of " + aliases + " skip locked"
+				: getForUpdateString( aliases );
+	}
+
+	@Override
+	public boolean supportsOuterJoinForUpdate() {
+		return false;
+	}
+
+	@Override
```

Comment:
Looks like something got lost after the rebase:

```suggestion
	@Override
	public boolean useInputStreamToInsertBlob() {
		return false;
	}

	@Override
```